src/Microsoft.CodeAnalysis.Analyzers/CSharp/MetaAnalyzers/Fixers/CSharpPreferIsKindFix.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.Fixers;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.Analyzers.MetaAnalyzers.Fixers
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(CSharpPreferIsKindFix))]
+    [Shared]
+    public sealed class CSharpPreferIsKindFix : PreferIsKindFix
+    {
+        protected override async Task<Document> ConvertKindToIsKindAsync(Document document, TextSpan sourceSpan, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var binaryExpression = root.FindNode(sourceSpan, getInnermostNodeForTie: true).FirstAncestorOrSelf<BinaryExpressionSyntax>();
```

Comment:
always good to have a test where this is an argument to ensure the `getInnermostNodeForTie:` is warranted.  (if you already have that test, aces).

Reply:
I wasn't able to find a case where it is needed.

Reply:
Do this:

`Foo(x.Kind() == SyntaxKind.Whatever`

With this, getInnermost is needed as otherwise FindNode will find the ArgumentSyntax.

Reply:
Argument vs expression is my go to case wherever I want to test that.

Reply:
This was my first thought, but the diagnostic is reported on the left child of the binary expression, not the full binary expression.

Reply:
ah, gotcha.  in that case, you're likely fine.